extensions/controllers/sandboxclaim_controller.go

```diff
@@ -79,6 +87,13 @@ func (r *SandboxClaimReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		// Try getting sandbox even if template is not found
 		// It is possible that the template was deleted after the sandbox was created
 		sandbox, err = r.getOrCreateSandbox(ctx, claim, template)
+
+		if err == nil { // Only reconcile NetworkPolicy if sandbox creation was successful
+			if npErr := r.reconcileNetworkPolicy(ctx, claim, template); npErr != nil {
```

Comment:
Security risk. This happens after `getOrCreateSandbox`, so pods could be running before the network policy is created.

Reply:
sounds good, switched the order